The Big Refactor

Amaras opened the pull request:
This PR aims to allow an easier understanding and ability to change the code in the future.
The changes are currently not ready to merge, and not tested yet.

This is a summary of the current state of changes (at the time of writing):
- [x] Used 3.10 syntax to remove Optional[T] and Union[T, U] in code
- [x] Refactor to minify the typing dependency in all the classes in `data/`
- [x] Changed some classes into dataclasses
- [x] Solved some mypy errors

The plan for the future is based on this talk by Conor Hoekstra: https://www.youtube.com/watch?v=nXZQfdxWgh0
In summary for our case:
- [x] Use Enum (especially for sets of related bare constants)
- [x] ~~Maybe use Optional (from optional.py)~~ decided against it because of no `__class_getitem__`
- [ ] Minimize the "Initialize Then Modify" pattern
- [x] (Not related to the talk) Use direct assignment instead of the `set_` methods when possible, and add warnings for the previous API.

There might be more refactors if applicable (for example: extract out the data classes creation to the classes as classmethods).

Amaras committed:
Used 3.10 syntax to remove Optional[T] and Union[T, U] in code

Amaras committed:
Forgot to change template class

Amaras committed:
Refactor to minify the typing dependency (uses 3.9 and 3.10 syntax)

simplified typing dependencies in data/event.py

Removed typing dependency in data/player.py

Removed typing dependency in data/result.py

Removed typing dependency in data/rotator.py

Removed typing dependency in data/screen.py

Simplified typing import in data/screen_set.py

Removed typing dependency in data/template.py

Removed typing dependency in data/timer.py

Removed typing import in data/tournament.py

Removed typing dependency in common/config_reader.py

Removed lint errors in common/engine.py

Removed lint error in common/logger.py

Removed typing dependency in common/papi_web_config.py

simplified typing import from database/access.py

Removed typping import in database/papi.py

Removed typing dependency in web/views.py

Amaras committed:
Changed some classes into dataclasses

Amaras committed:
Added dataclasses and solved some mypy errors

Amaras committed:
Refactored AccessDatabase into a dataclass

Amaras committed:
Merge branch 'main' into refactor_to_3.11

pascalaubry reviewed (commented):
Why did Player.__board_number disappear?

Amaras committed:
Added back Player.__board_number

Amaras committed:
Partial work on Enum refactor

Amaras committed:
Merge branch 'main' into refactor_to_3.11

Amaras committed:
Massive name simplification, along with other changes

Amaras committed:
Snapshot for computer transfer

pascalaubry committed:
Merge branch 'main' into refactor_to_3.11

Amaras committed:
Work from 2023-11-06

Amaras committed:
Merge branch 'main' into refactor_to_3.11

Amaras committed:
Merge remote-tracking branch 'origin/refactor_to_3.11' into refactor_to_3.11

# Conflicts:
#	papi_web.py
#	web/templates/screen_boards.html

pascalaubry committed:
Details

pascalaubry committed:
Fixed args.path
Renamed __database, __file and __cursor

Amaras committed:
Fixed missing players by name bug

Changed a few NOTE items into TODO items

pascalaubry committed:
Bug fix on White forfeit and Black present
Remark on Berger-paired tournaments

Amaras committed:
Moved ScreenType into its own class, and deleted previous constants

Also fixed inspection problems

Amaras committed:
Merge remote-tracking branch 'origin/refactor_to_3.11' into refactor_to_3.11

pascalaubry committed:
Changed super.enter() to super.__enter__() (no impact IMO)

Amaras committed:
Fixed input bug

pascalaubry committed:
Merge remote-tracking branch 'big-refactor/refactor_to_3.11' into refactor_to_3.11

Amaras committed:
Merge remote-tracking branch 'origin/refactor_to_3.11' into refactor_to_3.11

pascalaubry committed:
Loop on boards rather than on board ids

pascalaubry committed:
Commented TODOs

pascalaubry committed:
Added TODOs

pascalaubry committed:
Removed last line

pascalaubry committed:
Added TODOs to remember to delete variables PLAYER_TITLE_* and COLOR_*

pascalaubry committed:
Details

pascalaubry committed:
Added class TimerBuilder to lighten class Event

pascalaubry committed:
Renamed TimerBuilder.__config_reader to config_reader

pascalaubry committed:
Renamed TimerBuilder.__*() to _*()

Amaras committed:
Used the context-manager way of using the database

Also added a couple TODO items (about board ordering + unused (?) constants)

pascalaubry committed:
Added class RotatorBuilder to lighten class Event

pascalaubry committed:
Merge remote-tracking branch 'big-refactor/refactor_to_3.11' into refactor_to_3.11

pascalaubry committed:
Added class RotatorBuilder to lighten class Event

Amaras committed:
Removed unused constants

Also changed former TODOs into NOTES.
Added a short-circuit for board comparison.
Changed docs for TournamentPairing

Amaras committed:
Merge remote-tracking branch 'origin/refactor_to_3.11' into refactor_to_3.11

Amaras committed:
Removed all setters that threw UserWarnings

pascalaubry committed:
Added classes ScreenSetBuilder and ScreenBuilder to lighten class Event

pascalaubry committed:
Moved build_rotators() from event.py to rotator.py

pascalaubry committed:
Simplified event.py by adding properties to builders

pascalaubry committed:
Simplified class ScreenSetBuilder

pascalaubry committed:
Added class FamilyBuilder

pascalaubry committed:
Added class TemplateBuilder

pascalaubry committed:
Added class TemplateBuilder

pascalaubry committed:
Fixed TemplateBuilder.__init__()

pascalaubry committed:
Fixed TemplateBuilder.__init__()

pascalaubry committed:
Added class TournamentBuilder

pascalaubry committed:
Removed unnecessary properties

pascalaubry committed:
Simplified class Engine
